src/main/java/com/cloudant/http/interceptors/CookieInterceptor.java

```diff
@@ -121,6 +121,15 @@ private String getCookie(URL url) {
 
             HttpConnection conn = Http.POST(sessionURL, "application/x-www-form-urlencoded");
             conn.setRequestBody(sessionRequestBody);
+
+            //when we request the session we need all interceptors except this one
+
+            conn.requestInterceptors.addAll(context.connection.requestInterceptors);
+            conn.requestInterceptors.remove(this);
+            conn.responseInterceptors.addAll(context.connection.responseInterceptors);
+            conn.responseInterceptors.remove(this);
```

Comment:
Should we also guard against the possibility that a user may have manually added a cookie interceptor to the interceptors list in addition to the the automatically generated one?